File: backend/app/schemas/service_history.py
from datetime import date

from pydantic import BaseModel

from app.schemas.maintenance import MaintenanceType


class ServiceHistoryItem(BaseModel):
    id: str
    type: MaintenanceType
    description: str | None = None
    cost: float | None = None
    notes: str | None = None


class ServiceHistoryVisit(BaseModel):
    id: str
    vehicle_id: str
    service_date: date
    mileage: int
    workshop: str | None = None
    notes: str | None = None
    total_cost: float | None = None
    items: list[ServiceHistoryItem]

File: frontend/src/features/maintenance/types/serviceHistory.ts
import type { MaintenanceType } from "./maintenance";

export interface ServiceHistoryItem {
    id: string;
    type: MaintenanceType;
    description: string | null;
    cost: number | null;
    notes: string | null;
}

export interface ServiceHistoryVisit {
    id: string;
    vehicle_id: string;
    service_date: string;
    mileage: number;
    workshop: string | null;
    notes: string | null;
    total_cost: number | null;
    items: ServiceHistoryItem[];
}
